internal/controller/ols_console_reconciliator.go

```diff
@@ -0,0 +1,267 @@
+		err = r.Create(ctx, service)
+		if err != nil {
+			return fmt.Errorf("failed to create Console UI service: %w", err)
+		}
+		r.logger.Info("Console UI service created", "service", service.Name)
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to get Console UI service: %w", err)
+	}
+	err = r.Update(ctx, service)
+	if err != nil {
+		return fmt.Errorf("failed to update Console UI service: %w", err)
+	}
+
+	err = r.Client.Get(ctx, client.ObjectKey{Name: ConsoleUIServiceName, Namespace: cr.Namespace}, foundService)
+	if err != nil {
+		return fmt.Errorf("failed to get Console UI service: %w", err)
+	}
+
+	r.logger.Info("Console UI service reconciled", "service", service.Name)
+
+	return nil
+}
+func (r *OLSConfigReconciler) reconcileConsoleUIDeployment(ctx context.Context, cr *olsv1alpha1.OLSConfig) error {
+	deployment, err := r.generateConsoleUIDeployment(cr)
+	if err != nil {
+		return fmt.Errorf("failed to generate Console Plugin deployment: %w", err)
+	}
+	foundDeployment := &appsv1.Deployment{}
+	err = r.Client.Get(ctx, client.ObjectKey{Name: ConsoleUIDeploymentName, Namespace: cr.Namespace}, foundDeployment)
+	if err != nil && errors.IsNotFound(err) {
+		r.logger.Info("creating Console UI deployment", "deployment", deployment.Name)
+		err = r.Create(ctx, deployment)
+		if err != nil {
+			return fmt.Errorf("failed to create Console UI deployment: %w", err)
+		}
+		r.logger.Info("Console UI deployment created", "deployment", deployment.Name)
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to get Console UI deployment: %w", err)
+	}
+
+	if deploymentSpecEqual(&foundDeployment.Spec, &deployment.Spec) {
+		r.logger.Info("Console UI deployment unchanged", "deployment", deployment.Name)
+		return nil
+	}
+
+	err = r.Update(ctx, deployment)
+	if err != nil {
+		return fmt.Errorf("failed to update Console UI deployment: %w", err)
+	}
+	r.logger.Info("Console UI deployment reconciled", "deployment", deployment.Name)
+
+	return nil
+}
+
+func (r *OLSConfigReconciler) reconcileConsoleUIPlugin(ctx context.Context, cr *olsv1alpha1.OLSConfig) error {
+	plugin, err := r.generateConsoleUIPlugin(cr)
+	if err != nil {
+		return fmt.Errorf("failed to generate Console Plugin: %w", err)
+	}
+	foundPlugin := &consolev1.ConsolePlugin{}
+	err = r.Client.Get(ctx, client.ObjectKey{Name: ConsoleUIPluginName}, foundPlugin)
+	if err != nil && errors.IsNotFound(err) {
+		r.logger.Info("creating Console Plugin", "plugin", plugin.Name)
+		err = r.Create(ctx, plugin)
+		if err != nil {
+			return fmt.Errorf("failed to create Console Plugin: %w", err)
+		}
+		r.logger.Info("Console Plugin created", "plugin", plugin.Name)
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to get Console Plugin: %w", err)
+	}
+
+	if apiequality.Semantic.DeepEqual(foundPlugin.Spec, plugin.Spec) {
+		r.logger.Info("Console Plugin unchanged, skip reconciliaiton", "plugin", plugin.Name)
```

Comment:
Same here + typo. Must be "reconciliation"
Throughout the code we use ".... reconciliation skipped"